### PR TITLE
Tel 56 create delete tenant endpoint

### PR DESCRIPTION
**Pull request by AmitkumarEvry**
Delete tenant endpoint
unit test cases

### CODE REVIEW

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: badc9076580f20085d3505bcd283af8828c17887
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ad89bc49-e176-4637-8139-b6a65ef504d1%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by NagarajNidagundi** [comment]
Looks good

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: f3807a8547ad5a24f55d24e76ce512b1f718e9cd
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=99ee997f-5565-42cf-b0e5-4476f08d27c8%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.TenantService/Application/TenantManager.cs [hunk 1]
```diff
@@ -129,6 +135,34 @@ public ApplicationLoadBalancerResponse CreateTenant(string tenantRequest)
...
+            Tenant tenant = this.tenantContext.Tenants.Include(a => a.TenantKeys).Where(p => p.PublicId.Equals(publicId)).FirstOrDefault();
+
+            if (tenant == null)
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(404);
+            }
+
+            if (tenant.TenantKeys.Count > 0)
+            {
+                return ALBResponseHelper.GetDefaultALBResponse(409, "This tenant has tenantkey associated with it.");
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService/Application/TenantManager.cs]: The ClientGroup behavior was a special case.  We should delete all associated Tenant Keys here.

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: 8aa33ca417ab067fdbfb9dcc91b0a7f5886e2c51
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8ccc517c-2344-4c6b-8e07-e76050ab68f1%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: a4811f1c0feb30d57e1e599a70be1b709206ce56
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e8659971-04fa-44cc-bdb1-b355c56d10c1%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.TenantService/Application/TenantManager.cs (hunk 1 above)
Comment (AmitkumarEvry) [reply src/WWTelematics.TenantService/Application/TenantManager.cs]: Added cascasde delete in fluent api to delete all  child tenantkeys of tenant.

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: 93b1a0e64255f298a9a699f1d771c6502d378937
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=60c792fc-754c-431e-91e6-3aa27590a92f%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: 38d5f60df1ab6826a3736cf92aeaa1c29e8cb16a
* Result: FAILED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ed35d250-be99-41f9-b82b-0a527431599e%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**